Add OSM Scout Valhalla router

otsaloma opened the pull request:
@rinigus Can you check if this is OK?

* The router is added under the name "OSM Scout Valhalla". If I understood correctly from earlier discussions, we'd support both libosmscout and Valhalla for now and maybe drop libosmscout later?

* I marked "harbour-osmscout-server-module-route" as a requirement, so the Valhalla router is only visible to those who have the module installed.

* The "osmscout" geocoder is used to get coordinates.

otsaloma committed:
Add OSM Scout Valhalla router

rinigus commented:
Thank you very much for this addition! I run some tests and all seems to be working as it should.

* Name is, indeed, a tricky question. Yes, plan for a bit of time would be to support libosmscout via `v1` protocol and Valhalla via `v2`, that's right. Long-term target is to have just one "OSM Scout" as a router. This would require some work on my part and I have to see how hard is it, but I would probably try to move libosmscout over to Valhalla's protocol as well. Whether it would be implemented would all depend on amount of time and demand. So, probably we will rename, in the end, Valhalla's router into  "OSM Scout". Question is when and whether to do it now and call the previous version "OSM Scout Legacy"? It's a messy situation right now without any great solution. [ I guess the router name "OSM Scout Libosmscout" is rather long :) ]. However, combination "OSM Scout Valhalla" / "OSM Scout Legacy" would require some action on the part of the users which is probably adequate representation of the mess I created.

* Requirement - great idea! Maybe we should add `harbour-osmscout-server` as one more requirement if it does support multiple requirements. Since I cannot have any dependencies in RPMs, its possible to install `harbour-osmscout-server-module-route` without `harbour-osmscout-server`.

* Geocoder: that's exactly what's needed for offline navigation.

* In the last sentence of the displayed route, you have "Routing courtesy of %1". Since we know that the route is calculated by Valhalla, would be fair to stress it. Maybe something like `Routing courtesy of Valhalla via OSM Scout Server`?

otsaloma committed:
Add harbour-osmscout-server to requires

otsaloma committed:
Reword attribution

otsaloma commented:
> Question is when and whether to do it now and call the previous version "OSM Scout Legacy"?

Whichever way is fine by me, I'll leave that up to you. I think it's just important for the naming to be consistent so that users know what to download in the server app and what to use in Poor Maps. So, I can rename the libosmscout one "OSM Scout Legacy" if you too use "Legacy" in your server UI.

> Maybe we should add harbour-osmscout-server as one more requirement if it does support multiple requirements.

Added.

> In the last sentence of the displayed route, you have "Routing courtesy of %1". Since we know that the route is calculated by Valhalla, would be fair to stress it. Maybe something like Routing courtesy of Valhalla via OSM Scout Server?

That "courtesy of" refers to the service, that's how the commercial providers require attribution. I changed it to "Routing by Valhalla, courtesy of OSM Scout Server". This OK?

rinigus commented:
**Re naming**: Its a good point, I don't use "Legacy" in GUI. Taking into account the current complexity already, let's keep the new terms to the minimum. In addition, the two interfaces are temporary (I have added an issue https://github.com/rinigus/osmscout-server/issues/152 to fix it), so there is no need for users to learn it. So, I suggest to have the names:

OSM Scout Valhalla
OSM Scout libosmscout

These are terms that are used in GUI and users should be familiar with that. Leaving a plain `OSM Scout` would probably just bring out more confusion. Meanwhile, I will work on translating Valhalla's protocol to make it usable by libosmscout. It should not take too much time, I think. But, ideally, I would do it a touch later (working on Nearby search in geocoder-nlp right now).

**Re "courtesy of"**: Looks good to me.

otsaloma committed:
Add "libosmscout" to legacy router name

otsaloma commented:
Thanks, merging. I'll make a release within the next few days.

rinigus commented:
Thank you very much!
